best-practice-on-public-cloud.md

```diff
@@ -0,0 +1,182 @@
+
+### Example 1, Run a social network workload on AWS
+With 20GB [gp3](https://aws.amazon.com/ebs/general-purpose/) dedicated Raft-Engine disk, for a write intensive social network application workload, qps increases by 17.5%, avg latency of insert statement decreases by 18.7%, p99 latency of insert statement decreases by 45.6%, and the estimated cost is increased by only 0.4%. AWS provide 3000 IOPS and 125 MBPS/s for a 20GB [gp3](https://aws.amazon.com/ebs/general-purpose/) volume.
+
+| Item | shared Raft-Engine disk|dedicated Raft-Engine disk| diff(%) |
+| ------------- | ------------- |------------- |------------- |
+| QPS (K/s)| 8.0 | 9.4 | 17.5| 
+| AVG Insert Latency (ms)| 11.3 | 9.2 | -18.7 |
+| P99 Insert Latency (ms)| 29.4 | 16.0 | -45.6|
+
+### Example 2, Run TPC-C/SYSBench workload on AZure
+By using a 32G Ultra disk for Raft-Engine on Azure:
+
+- For sysbench oltp_read_write,  QPS increases by 17.8%, avg latency decreases by 15.6%.
+- For TPC-C, QPS increased by 27.6%, avg latency decreases by 23.1%
+
+| Item | Workload | shared Raft-Engine disk|dedicated Raft-Engine disk| diff(%) |
+| ------------- | ------------- | ------------- |------------- |------------- |
+| QPS (K/s) | Sysbench - oltp_read_write | 60.7 | 71.5 | 17.8| 
+| QPS (K/s) | TPC-C | 23.9 | 30.5 | 27.6| 
+| AVG Latency (ms)| Sysbench - oltp_read_write |  4.5 | 3.8 | -15.6 |
+| AVG Latency (ms)| TPC-C |  3.9 | 3.0 | -23.1 |
+
+### Example 3, how to attach a dedicated pd-ssd disk on Google Cloud for Raft-Engine on TiKV manifest
+
+Here is a example for a cluster deployed by TiDB Operator on Google Cloud, attach an additional disk with 512GB [pd-ssd](https://cloud.google.com/compute/docs/disks#disk-types/) volumne, and change the `raft-engine.dir` to store Raft-Engine logs to this specific disk.
+
+```
+tikv:
+    config: |
+      [raft-engine]
+        dir = "/var/lib/raft-pv-ssd/raft-engine"
+        enable = true
+        enable-log-recycle = true
+    requests:
+      storage: 4Ti
+    storageClassName: pd-ssd
+    storageVolumes:
+    - mountPath: /var/lib/raft-pv-ssd
+      name: raft-pv-ssd
+      storageSize: 512Gi
+```
+
+
+## Reduce Compaction IO Flow In KV RocksDB
+As the storage engine of TiKV, [RocksDB](https://rocksdb.org/) is used to store user data. The write amplification for RocksDB can be high and the workload may be bottlenecked on disk throughput. This is not uncommon because the provisioned IO throughput on cloud ebs is usually limited due to cost reason. Under this situation, the total number of compaction pending bytes will grow over time and flow control is triggered, indicating that TiKV doesn't have enough disk bandwith to keep up with the foreground write flow. In this case, to mitigate the bottleneck of disk throughput, increasing the compression level for the RocksDB and reducing the disk throughput can help improve performance. For example, below config increase all the compression level of the default column family to zstd.
```

Comment:
```suggestion
As the storage engine of TiKV, [RocksDB](https://rocksdb.org/) is used to store user data. However, because the provisioned IO throughput on cloud EBS is usually limited due to cost considerations, RocksDB might exhibit high write amplification, and the disk throughput might become the bottleneck for the workload. As a result, the total number of pending compaction bytes grows over time and triggers flow control, which indicates that TiKV lacks sufficient disk bandwidth to keep up with the foreground write flow. 

To alleviate the bottleneck caused by limited disk throughput, you can improve performance by increasing the compression level for RocksDB and reducing the disk throughput. For example, you can refer to the following example to increase all the compression levels of the default column family to `zstd`.
```